### PR TITLE
add suggester related meters

### PR DESCRIPTION
**Pull request by vladak**
Add basic suggester meters:
```
# HELP suggester_init_latency_seconds suggester initialization latency
# TYPE suggester_init_latency_seconds summary
suggester_init_latency_seconds_count 1.0
suggester_init_latency_seconds_sum 0.035
# HELP suggester_init_latency_seconds_max suggester initialization latency
# TYPE suggester_init_latency_seconds_max gauge
suggester_init_latency_seconds_max 0.035
# HELP suggester_rebuild_latency_seconds_max suggester rebuild latency
# TYPE suggester_rebuild_latency_seconds_max gauge
suggester_rebuild_latency_seconds_max 0.0
# HELP suggester_rebuild_latency_seconds suggester rebuild latency
# TYPE suggester_rebuild_latency_seconds summary
suggester_rebuild_latency_seconds_count 0.0
suggester_rebuild_latency_seconds_sum 0.0
# HELP suggester_rebuild_total suggester rebuild count
# TYPE suggester_rebuild_total counter
suggester_rebuild_total 0.0
```

### CODE REVIEW

File: suggester/src/main/java/org/opengrok/suggest/Suggester.java [hunk 1]
```diff
@@ -124,8 +134,8 @@ public Suggester(
             final boolean projectsEnabled,
             final Set<String> allowedFields,
             final int timeThreshold,
-            final int rebuildParallelismLevel
-    ) {
+            final int rebuildParallelismLevel,
+            MeterRegistry registry) {
```
Comment (ahornace) [inline suggester/src/main/java/org/opengrok/suggest/Suggester.java:138]: Hmm, it seems we are facing telescoping constructor anti-pattern. I'll try to think of something (builder, etc.). Feel free to ignore for now.

**Review by ahornace — CHANGES_REQUESTED** [review]

File: suggester/src/main/java/org/opengrok/suggest/Suggester.java [hunk 2]
```diff
@@ -97,6 +100,12 @@
 
     private final CountDownLatch initDone = new CountDownLatch(1);
 
+    private final MeterRegistry registry;
```
[the remark this answers is not in the document]
Comment (vladak) [reply suggester/src/main/java/org/opengrok/suggest/Suggester.java]: fixed

File: suggester/src/main/java/org/opengrok/suggest/Suggester.java (hunk 1 above)
Comment (vladak) [reply suggester/src/main/java/org/opengrok/suggest/Suggester.java:136]: okay, ignoring :-)

File: suggester/src/main/java/org/opengrok/suggest/Suggester.java [hunk 3]
```diff
@@ -143,6 +151,16 @@ public Suggester(
         this.allowedFields = new HashSet<>(allowedFields);
         this.timeThreshold = timeThreshold;
         this.rebuildParallelismLevel = rebuildParallelismLevel;
+
+        suggesterRebuildCounter = Counter.builder("suggester.rebuild").
+                description("suggester rebuild count").
+                register(registry);
+        suggesterRebuildTimer = Timer.builder("suggester.rebuild.latency").
```
Comment (ahornace) [inline suggester/src/main/java/org/opengrok/suggest/Suggester.java:158]: Now that I think about it. Does not `Timer` include also a `Counter` by default? In the code it seems they do: https://github.com/micrometer-metrics/micrometer/blob/master/implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/PrometheusTimer.java

Could you please check if you can plot it in Grafana?

Comment (vladak) [reply suggester/src/main/java/org/opengrok/suggest/Suggester.java:158]: Indeed it does. I wanted to mention it previously that I'd like to keep it separate.

```
$ curl -X GET -s http://grok-stage.cz.oracle.com/source/metrics/prometheus | grep ^suggest
suggester_init_latency_seconds_count 1.0
suggester_init_latency_seconds_sum 0.035
suggester_init_latency_seconds_max 0.0
suggester_rebuild_latency_seconds_max 0.0
suggester_rebuild_latency_seconds_count 30.0
suggester_rebuild_latency_seconds_sum 0.012
suggester_rebuild_total 30.0
```

Comment (ahornace) [reply suggester/src/main/java/org/opengrok/suggest/Suggester.java:158]: Okay, no problem :)

**Pull request merged**